### DISCUSSION TITLE
Wrong mapping

### DISCUSSION THREAD

**Discussion by aplasa85** [neuefische/questions#375] [Web] [answered]
Hi, 

I have an issue with the mapping of my pictures. It seems as there is a definition problem but I cannot seem to figure out what it is. Seemingly, the code should be working as it does not throw any other error, but it does not render nor does it recognise the databank of TequilaList. 

```js
import React from "react";
import Head from "next/head";
import Image from "next/image"
import Icon from "../public/Assets/Agave.jpg"
//import TequilaCard from "./TequilaCard"
import TequilaList from "../_data/TequilaList"
import styled from "styled-components"
import Link from "next/link";
export default Tequila

const tequila = () => {
  return (
<>
<Head>
      <div>
        <title>Tequila</title>
      </div>
     
    </ Head>

```

-------

```js
    

const TequilaCard = ({ tequila }) => {
   (



      <Link href="/product/[id]" as={`/product/${tequila.id}`}>
          <StyledCard>

              <h3> {tequila.brand} &rarr; </h3>
              <p>{tequila.type}</p>
              <p>{tequila.rating}</p>
              <p>{tequila.price}</p>
              {
                  <image src={`../public/Assets/TequilaPics`} width="150" height="150" alt="an image" />
              }


          </StyledCard>
      </Link>
      
 )
Thanks in advance for your help. 
```

https://github.com/aplasa85/capstone-template-ffm-web-22-1

**Comment by actyvystom** [comment]
Hi Axel,

please narrow down your problem to relevant code parts and describe your problem in more detail. A link to your project repo does not help - [see guidelines for creating questions](https://github.com/orgs/neuefische/discussions/11)

**Comment by actyvystom** [comment]
Just from a quick glance: 
- your img `src` does not contain a valid image path (it's just a folder)
- there is no mapping at all in your code so what exactly do you mean by *mapping of your pictures*? 
- Your TequilaCard component does not return anything, so there is probably no html output at all
- Your code looks really messed up, maybe you missed to post some parts?
- Again, please try to narrow down your problem

**Comment by aplasa85** [comment, marked as answer]
Hi Thomas,

Indeed, there has been some editing done to he code, due to some eslint
suggestions. The only way to satisfy the beast is to mutilate the whole
code. Otherwise, I did have the mapping before as I can remember, and it
ended looking like this more or less:


const TequilaList = ({ tequila }) => {
(



<Link href="/product/[id]" as={`/product/${tequila.id}`}>
<StyledCard>
{tequila.map} => {
return (Tequila.id))

<h3> {tequila.brand} &rarr; </h3>
<p>{tequila.type}</p>
<p>{tequila.rating}</p>
<p>{tequila.price}</p>
{
<image src={`../public/Assets/TequilaPics`} width="150" height="150" alt="an
image" />
}
}

</StyledCard>
</Link>
)


Nonetheless, again there is no mapping happening here for some reason. Will
try to approach this problem today as soon as we start our session.
